### PR TITLE
Updating FocusTrapZone, Overlay, and TextField examples to use function components

### PR DESCRIPTION
**Pull request by czearing**
#### Pull request checklist

- [x] Include a change request file using `$ yarn change`

#### Description of changes

Updated FocusTrapZone, Overlay, and TextField examples to use function components

#### Focus areas to test

- FocusTrapZone
- Overlay
- TextField

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12774)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/TextField/examples/TextField.CustomRender.Example.tsx [hunk 1]
```diff
@@ -25,96 +26,84 @@ const getDescriptionStyles = memoizeFunction((theme: ITheme) => ({
...
-          label="Custom description rendering"
-          description="A colorful description!"
-          onRenderDescription={this._onRenderDescription}
+  return (
+    <>
+      <Stack horizontal verticalAlign="center" tokens={{ childrenGap: 4 }}>
+        <span id={labelId}>{props.label}</span>
+        <IconButton
+          id={iconButtonId}
+          iconProps={{ iconName: 'Info' }}
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/TextField/examples/TextField.CustomRender.Example.tsx:65]: move this to a const

File: packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Controlled.Example.tsx [hunk 2]
```diff
@@ -5,49 +5,40 @@ import { Stack } from 'office-ui-fabric-react/lib/Stack';
...
+  ) => {
+    setFirstTextFieldValue(newValue || '');
   };
-
-  private _onChange2 = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
+  const onChangeSecondTextFieldValue = (
+    event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+    newValue?: string,
+  ) => {
     if (!newValue || newValue.length <= 5) {
```
[the remark this answers is not in the document]
Comment (czearing) [reply packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Controlled.Example.tsx]: @ecraig12345, thanks for reviewing! I ran TextField.Controlled.Example and FocusTrapZone.Nested.Example locally and they appear to run the same as before.

File: packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx [hunk 3]
```diff
@@ -2,17 +2,16 @@ import * as React from 'react';
...
 
+const stackTokens = { childrenGap: 50 };
 const stackStyles: Partial<IStackStyles> = { root: { width: 650 } };
+const columnProps: Partial<IStackProps> = {
+  tokens: { childrenGap: 15 },
+  styles: { root: { width: 300 } },
+};
 
 export const TextFieldBasicExample: React.FunctionComponent = () => {
-  // TextFields don't have to be inside Stacks, we're just using Stacks for layout
```
Comment (khmakoto) [inline packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx:8]: Could you re-add this comment that was lost in translation?

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx:8]: My 2c; I'd think it's probably ok to leave it out and keep the example simple, seems obvious they're there for layout. We use Stack elsewhere without the clarification.

File: packages/office-ui-fabric-react/src/components/FocusTrapZone/examples/FocusTrapZone.Box.FocusOnCustomElement.Example.tsx [hunk 4]
```diff
@@ -16,61 +17,40 @@ const getStackStyles = memoizeFunction(
...
-    this.setState({
-      useTrapZone: true,
-    });
-  };
-
-  private _onFocusTrapZoneToggleChanged = (ev: React.MouseEvent<HTMLElement>, checked?: boolean): void => {
-    this.setState({ useTrapZone: !!checked });
-  };
-}
+const toggle = React.createRef<IToggle>();
```
[the remark this answers is not in the document]
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/FocusTrapZone/examples/FocusTrapZone.Box.FocusOnCustomElement.Example.tsx:27]: Great catches!

File: packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx (hunk 3 above)
Comment (khmakoto) [reply packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx:8]: Sounds good to me, I'll resolve this comment.

File: packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx [hunk 5]
```diff
@@ -21,12 +20,9 @@ export const TextFieldBasicExample: React.FunctionComponent = () => {
         <TextField ariaLabel="Required without visible label" required />
         <TextField label="With error message" errorMessage="Error message" />
       </Stack>
-
       <Stack {...columnProps}>
         <MaskedTextField label="With input mask" mask="m\ask: (999) 999 - 9999" />
-
         <TextField label="With an icon" iconProps={{ iconName: 'Calendar' }} />
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx:25]: Can you move this into a constant too?

**Pull request merged**